spec/requests/api/rest/admin/gateways_spec.rb

```diff
@@ -433,4 +460,236 @@
+        'single-codec-in-200ok': gateway.single_codec_in_200ok,
+        'transparent-seqno': gateway.transparent_seqno,
+        'transparent-ssrc': gateway.transparent_ssrc,
+        'force-symmetric-rtp': gateway.force_symmetric_rtp,
+        'symmetric-rtp-nonstop': gateway.symmetric_rtp_nonstop,
+        'symmetric-rtp-ignore-rtcp': gateway.symmetric_rtp_ignore_rtcp,
+        'force-dtmf-relay': gateway.force_dtmf_relay,
+        'rtp-ping': gateway.rtp_ping,
+        'rtp-timeout': gateway.rtp_timeout,
+        'filter-noaudio-streams': gateway.filter_noaudio_streams,
+        'rtp-relay-timestamp-aligning': gateway.rtp_relay_timestamp_aligning,
+        'rtp-force-relay-cn': gateway.rtp_force_relay_cn,
+        'incoming-auth-username': gateway.incoming_auth_username,
+        'incoming-auth-password': gateway.incoming_auth_password
+      }
+    end
+
+    let(:json_api_request_path) { "#{super()}/#{record_id}" }
+    let(:request_params) { nil }
+    let(:record_id) { gateway.id.to_s }
+
+    let!(:gateway) { FactoryBot.create(:gateway) }
+
+    include_examples :returns_json_api_record, relationships: gateway_relationship_names do
+      let(:json_api_record_id) { record_id }
+      let(:json_api_record_attributes) { gateway_response_attributes }
+    end
+  end
+
+  describe 'POST /api/rest/admin/gateways' do
+    subject do
+      post json_api_request_path, params: json_api_request_body.to_json, headers: json_api_request_headers
+    end
+
+    let(:contractor) { FactoryBot.create(:customer) }
+    let(:session_refresh_method) { SessionRefreshMethod.first }
+    let(:sdp_alines_filter_type) { FilterType.first }
+    let(:codec_group) { FactoryBot.create(:codec_group) }
+    let(:sdp_c_location) { SdpCLocation.first }
+    let(:sensor_level) { System::SensorLevel.first }
+    let(:dtmf_receive_mode) { System::DtmfReceiveMode.first }
+    let(:dtmf_send_mode) { System::DtmfSendMode.first }
+    let(:transport_protocol) { Equipment::TransportProtocol.first }
+    let(:term_proxy_transport_protocol) { Equipment::TransportProtocol.first }
+    let(:orig_proxy_transport_protocol) { Equipment::TransportProtocol.first }
+    let(:rel100_mode) { Equipment::GatewayRel100Mode.first }
+    let(:rx_inband_dtmf_filtering_mode) { Equipment::GatewayInbandDtmfFilteringMode.first }
+    let(:tx_inband_dtmf_filtering_mode) { Equipment::GatewayInbandDtmfFilteringMode.first }
+    let(:network_protocol_priority) { Equipment::GatewayNetworkProtocolPriority.first }
+    let(:media_encryption_mode) { Equipment::GatewayMediaEncryptionMode.first }
+    let(:sip_schema) { System::SipSchema.first }
+
+    let(:json_api_request_attributes) do
+      {
+        name: 'Test name',
+        enabled: false,
+        priority: 100,
+        weight: 100,
+        'acd-limit': 0.0,
+        'asr-limit': 0.0,
+        host: 'test.example.com',
+        'incoming-auth-username': 'incoming_username',
+        'incoming-auth-password': 'incoming_password'
```

Comment:
do not add new attributes here

youre POST describe should look like this

```ruby
  describe 'POST /api/rest/admin/gateways' do
    # ...

    let(:json_api_request_attributes) do
      # only required attributes
    end

    include_examples :returns_json_api_record, relationships: gateway_relationship_names, status: 201 do
      let(:json_api_record_id) { last_gateway.id.to_s }
      let(:json_api_record_attributes) do
        hash_including(json_api_request_attributes)
      end
    end

    it 'creates gateway with correct attributes' do
      expect { subject }.to change { Gateway.count }.by(1)

      expect(last_gateway).to have_attributes(
        # ...
      )
    end

    context 'with incoming-auth-username and incoming-auth-password' do
      let(:json_api_request_attributes) do
  	super().merge 'incoming-auth-username': 'incoming_username',
                      'incoming-auth-password': 'incoming_password'
      end

      include_examples :returns_json_api_record, relationships: gateway_relationship_names, status: 201 do
        let(:json_api_record_id) { last_gateway.id.to_s }
        let(:json_api_record_attributes) do
          hash_including(json_api_request_attributes)
        end
      end

      it 'creates gateway with correct attributes' do
        expect { subject }.to change { Gateway.count }.by(1)

        expect(last_gateway).to have_attributes(
          # ...
        )
      end
    end

  end
```

Reply:
because we need to keep test where only required attributes are passed - to ensure that you can create gateway without passing new attributes

Reply:
we will not add new context on PATCH describe because in case of PATCH we need only to test that attributes are updatable